src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -101,4 +168,10 @@ void invalidateChannelIfSequencersMatch(
   int getSize() {
     return cache.size();
   }
+
```

Comment:
1. javadoc missing
2. can you place this method next to the iterator() method?
3. Once you expose the entrySet(), the caller can just call .iterator() on top of the returned entrySet. Thus can probably remove the iterator() method from this class now.
4. The exposed Set<> is mutable. We should not allow callsites to mutate the internal state of this class; Can you wrap the returned object in a `Collections.unmodifiableSet(cache.entrySet())` ?